politeiawww/user/user.go

```diff
@@ -352,6 +352,41 @@ func (u *User) ActivateIdentity(key []byte) error {
 	return nil
 }
 
+// EmailHistory keeps track of the received emails by each user. This is
+// used to rate limit the amount of emails an user can receive in a 24h
+// time window. This was not stored in the user object in order to avoid
+// race conditions on db calls, since our user db currently does not support
+// transactions, and email notifications run in a separate goroutine. This
+// workaround won't be necessary once the user layer gets rewritten.
+type EmailHistory struct {
+	Timestamps       []int64 `json:"timestamps"`
+	LimitWarningSent bool    `json:"limitwarningsent"`
+}
+
+const VersionEmailHistory uint32 = 1
+
+// EncodeEmailHistory encodes EmailHistory into a JSON byte slice.
+func EncodeEmailHistory(h EmailHistory) ([]byte, error) {
```

Comment:
Not sure I follow this one mate. The encode/decode functions for the structs in this package are used to handle their respective blobs that we save and retrieve from the database. You mean we should do this inline where needed, instead of having these helper functions to do it?

Reply:
Yeah, the behavior of the helper functions is the exact same as using the json marshal/unmarshal methods, so why not just use them directly instead of creating a wrapper that doesn't add any additional functionality?

Reply:
Makes sense mate, will do